### ISSUE TITLE
Having "Force Integer Scaling" enabled during first game boot crashes medusa

### ISSUE THREAD

**Issue by yuriks** [mgba-emu/mgba#796]
Tested on eec3723.

Having "Force Integer Scaling" enabled during the first game boot in a session causes medusa (on Windows) to crash.

This can be worked around by temporarily disabling the setting before booting a game, and can then be re-enabled. This also doesn't occur on boots after the first one.

Config files triggering the crash: [medusa-force-integer-scaling-crash-config.zip](https://github.com/mgba-emu/mgba/files/1144153/medusa-force-integer-scaling-crash-config.zip)

**Comment by endrift** [comment]
Can't reproduce on macOS on fe797de35ee8a16ab49bd4ff77395b76b00c4b2c. Can you try updating your build?

**Comment by yuriks** [comment]
Still happens. Crash site/reason is:

```
Thread 1 received signal SIGFPE, Arithmetic exception.
0x0000000000466bf6 in QGBA::WindowBackground::paintEvent(QPaintEvent*) ()
```
(The stack trace doesn't have anything useful, it's just Qt frames.)

**Comment by endrift** [comment]
Can you try applying this patch and seeing if it helps? I'm on a different branch so you may need to apply it a bit manually:

```diff
diff --git a/src/platform/qt/Window.cpp b/src/platform/qt/Window.cpp
index 4c480a4..33af68b 100644
--- a/src/platform/qt/Window.cpp
+++ b/src/platform/qt/Window.cpp
@@ -151,7 +151,6 @@ Window::Window(ConfigController* config, int playerId, QWidget* parent)
 		QPixmap pixmap;
 		pixmap.convertFromImage(currentImage);
 		m_screenWidget->setPixmap(pixmap);
-		m_screenWidget->setLockAspectRatio(width, height);
 	});
 	connect(m_controller, &GameController::gamePaused, m_display, &Display::pauseDrawing);
 #ifndef Q_OS_MAC
@@ -740,6 +739,7 @@ void Window::gameStarted(mCoreThread* context, const QString& fname) {
 	context->core->desiredVideoDimensions(context->core, &width, &height);
 	m_display->setMinimumSize(width, height);
 	m_screenWidget->setMinimumSize(m_display->minimumSize());
+	m_screenWidget->setLockAspectRatio(width, height);
 	m_config->updateOption("lockIntegerScaling");
 	if (m_savedScale > 0) {
 		resizeFrame(QSize(width, height) * m_savedScale);
```

**Comment by yuriks** [comment]
Yep, that patch successfully fixes the issue. 👍

**Comment by yuriks** [comment]
Confirmed fixed by (I assume) 9ed7c91.
